packages/vaadin-virtual-list/src/vaadin-virtual-list.js

```diff
@@ -127,14 +127,32 @@ class VirtualListElement extends ElementMixin(ThemableMixin(PolymerElement)) {
 
   /** @private */
   __updateElement(el, index) {
+    if (el.__renderer !== this.renderer) {
+      el.__renderer = this.renderer;
+      this.__clearRenderTargetContent(el);
+    }
+
     if (this.renderer) {
       this.renderer(el, this, { item: this.items[index], index });
     }
   }
 
+  /**
+   * Clears the content of a render target.
+   * @private
+   */
+  __clearRenderTargetContent(element) {
+    element.innerHTML = '';
+    // Whenever a Lit-based renderer is used, it assigns a Lit part to the node it was rendered into.
+    // When clearing the rendered content, this part needs to be manually disposed of.
+    // Otherwise, using a Lit-based renderer on the same node will throw an exception or render nothing afterward.
+    delete element._$litPart$;
+  }
+
   /** @private */
   __itemsOrRendererChanged(items = [], renderer, virtualizer) {
-    if (renderer && virtualizer) {
+    const hasRenderedItems = this.childElementCount > 0;
```

Comment:
I'm about to approve this PR. 

However, I'm not sure if I fully understand what is the reason behind checking the existence of child elements?

Reply:
If `renderer` is not defined initially, the virtual list doesn't create elements at all even if it has `items` (there's a test for it also).

If you have a renderer initially (elements have been created) but remove it later, we still need to enter this logic to have the already existing items cleared.

Reply:
Oh, now I see, thanks! 👍

Reply:
Added a code comment for this to clarify the reason